### ISSUE TITLE
command palette shouldn't show actions that don't apply in the current context

### ISSUE THREAD

**Issue by hferreiro** [zed-industries/zed#4446]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

The command palette should filter the available actions depending on the context. For example, all of the "search: .." commands, shouldn't be shown outside of the search dialog.

### Environment

Zed: v0.105.5 (stable)
OS: macOS 13.5.2
Memory: 32 GiB
Architecture: aarch64

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.
If you only need the most recent lines, you can run the `zed: open log` command palette action to see the last 1000.

_No response_

**Comment by olejorgenb** [comment]
I agree, but find hiding actions like this confusing if my mental model of whether an action apply does not agree with reality. 

I suggest to sort inapplicable actions last and gray them out instead.

**Comment by JosephTLyons** [comment]
Hello,

We’re closing this issue as part of a cleanup of older feature requests.

Going forward, new feature requests start as GitHub Discussions and only become Issues once the direction is agreed on. This helps ensure that Issues represent scoped, ready-to-build work and makes it easier for contributors to find work that’s ready to pick up. You can read more in our [Feature Request Guidelines](https://github.com/zed-industries/zed/discussions/47963).

As part of this transition, we’re closing feature request Issues that are older than 6 months and have fewer than 10 upvotes.

If this feature hasn’t been implemented and is still important to you, we encourage you to open a new feature request Discussion [here](https://github.com/zed-industries/zed/discussions/new?category=feature-requests). That’s where we track and evaluate new requests.

Thanks for helping shape the future of Zed.
